galsim/utilities.py

```diff
@@ -296,6 +296,25 @@ def _convertPositions(pos, units, func):
 
     return pos
 
+def _lin_approx_err(x, f, i):
+    """Error as \int abs(f(x) - approx(x)) when using ith data point to make piecewise linear
+    approximation."""
+    xleft, xright = x[:i+1], x[i:]
+    fleft, fright = f[:i+1], f[i:]
+    xi, fi = x[i], f[i]
+    mleft = (fi-f[0])/(xi-x[0])
+    mright = (f[-1]-fi)/(x[-1]-xi)
+    f2left = f[0]+mleft*(xleft-x[0])
+    f2right = fi+mright*(xright-xi)
+    return np.trapz(np.abs(fleft-f2left), xleft), np.trapz(np.abs(fright-f2right), xright)
+
+def _lin_approx_split(x, f):
+    """Optimally split a tabulated function into two-part piecewise linear approximation.
+    """
+    errs = [_lin_approx_err(x, f, i) for i in xrange(1, len(x)-1)]
```

Comment:
I just noticed that this function is already O(N^2).  Which means that the full algorithm is actually O(N^3), since this gets called O(N) times.  It might be worth at least thinking a bit about whether this function could be made to be O(N).  I'll play around with it to see if I can come up with something.  Should be possible...